terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -662,6 +662,29 @@ def remove_inconsequential_statements(statements: List[JSON]) -> List[JSON]:
                 }
             }
         },
+        'aws_vpc_endpoint': {
```

Comment:
I think we determined that we could share one VPC endpoint and this change is based on that determination. However, one endpoint costs 1 cent/h so $7.20/month. Knowing that private deployments will only be needed until we have compliance, and knowing that we'd be adding even more infrastructure to get custom domain names working, we should move these resources to the Azul deployment, outside of the GitLab component.

You will need to lookup the VPC and the subnets via TF data sources but create the endpoint and a separate security group in each deployment.

The demo scenario is likely going to be switching a personal deployment to private and back.